### PR TITLE
Setting up yargs instead of meow

### LINKED ISSUE #428: Address dependency tree size
* [ ] [yargs](http://yargs.js.org/) is already available in g3. do we want to move to it and leave meow?  
* [x] do we need to address babar/loud-rejection/meow within speedline? possible to spoof via shrinkwrap, but unclear if that's even neccessary.
* [x] `devtools-timeline-model` has to match the version speedline uses.

### PR DESCRIPTION
**Pull request by wardpeet**
Moved meow to yargs.
I used variables because it looked better than chaining but perhaps you rather have chaining :)
Should fix #428

### CODE REVIEW

**Comment by brendankenny** [comment]
thanks for the PR!

I have a couple of notes I'll add, but they mostly boil down to preferring chaining :)

File: package.json [hunk 1]
```diff
@@ -39,7 +32,7 @@
     "coveralls": "^2.11.9"
   },
   "dependencies": {
-    "meow": "^3.7.0"
+    "yargs": "^4.7.1"
```
Comment (brendankenny) [inline package.json]: for a couple of reasons we need to use yargs at version 3.30.0 unless there were security/feature fixes that we really need

File: package.json [hunk 2]
```diff
@@ -8,25 +8,18 @@
     "node": ">=5"
   },
   "scripts": {
-    "//":
```
Comment (brendankenny) [inline package.json]: looks like npm autoformatted this file. Can you revert the changes to the `scripts` block?

File: lighthouse-cli/index.js [hunk 3]
```diff
@@ -32,41 +32,62 @@ if (semver.lt(process.version, '5.0.0')) {
...
+
+// Set option groups
+[
+  'Logging',
+  'Configuration',
+  'Output'
+].forEach(key => cliGroup(cli, `${key}:`, cliOptions[key]));
+
+// Set boolean values
+[
```
Comment (brendankenny) [inline lighthouse-cli/index.js]: I believe you can just do
```cli.boolean([
  'save-assets',
  'save-artifacts',
  'list-all-audits',
  'verbose',
  'quiet',
  'help',
  'version'
]);```

File: lighthouse-cli/index.js [hunk 4]
```diff
@@ -77,11 +98,11 @@ if (cli.flags.listAllAudits) {
   process.exit(0);
 }
 
-const url = cli.input[0] || 'https://m.aliexpress.com/';
-const outputMode = cli.flags.output || Printer.OUTPUT_MODE.pretty;
-const outputPath = cli.flags.outputPath || 'stdout';
-const flags = cli.flags;
-const config = (cli.flags.configPath && require(cli.flags.configPath)) || null;
+const url = cli._[0];
```
Comment (brendankenny) [inline lighthouse-cli/index.js]: need the default url in case one isn't supplied

File: lighthouse-cli/index.js [hunk 5]
```diff
@@ -32,41 +32,62 @@ if (semver.lt(process.version, '5.0.0')) {
...
-`, {
-  // These options do not have a value
-  boolean: [
-    'save-assets', 'save-artifacts', 'list-all-audits',
-    'verbose', 'quiet', 'help', 'version'
-  ]
-});
-
-if (cli.flags.listAllAudits) {
+const cliOptions = {
```
Comment (brendankenny) [inline lighthouse-cli/index.js]: can you collapse this and the `forEach`/`cliGroup` calls into a direct chain of yargs calls? More verbose, but we don't have very many options so it shouldn't be too bad (and nice to have it all laid out directly)

**Comment by brendankenny** [comment]
reviewed!

@wardpeet, do you think any of the other yargs features are worth using? `choices()` for `--output`, for instance

**Comment by paulirish** [comment]
hero!! thanks for this!!

File: package.json (hunk 2 above)
Comment (wardpeet) [inline package.json]: I thought it was pretty weird, i'll fix it ;)

File: package.json (hunk 1 above)
Comment (wardpeet) [inline package.json]: I'm just curious why wouldn't you upgrade to the latest version?

File: lighthouse-cli/index.js (hunk 3 above)
Comment (wardpeet) [inline lighthouse-cli/index.js]: indeed, just saw it as well :)

File: lighthouse-cli/index.js (hunk 4 above)
Comment (wardpeet) [inline lighthouse-cli/index.js]: than I have to strip 
`.demand(1, 'Please provide an url')`

**Comment by wardpeet** [comment]
@brendankenny I believe choices is a handy one and perhaps [verbose](https://github.com/yargs/yargs/tree/v3.30.0#yargs-even-counts-your-booleans) if you want multiple levels of logging?

File: package.json (hunk 1 above)
Comment (brendankenny) [inline package.json]: > I'm just curious why wouldn't you upgrade to the latest version?

We need to be on the same version as some other modules using yargs for an environment that can't use npm for module management.

**Comment by brendankenny** [comment]
what it looks like:

screen shot 2016-06-22 at 16 30 29

File: lighthouse-cli/index.js (hunk 4 above)
Comment (brendankenny) [inline lighthouse-cli/index.js]: mostly laziness, I think. It's nice for development and for showcasing without having a specific site in mind to test, and easy to catch if you need to run lighthouse against a specific url

**Comment by brendankenny** [comment]
sorry for the delay. This looks good to me except for keeping the default url. Thanks for taking this on!

**Comment by wardpeet** [comment]
ok than i don't force an url :) will update later today.

**Comment by paulirish** [comment]
We don't need a default URL anymore. It was nice, but we can avoid it now.

File: package.json [hunk 6]
```diff
@@ -39,7 +39,7 @@
     "coveralls": "^2.11.9"
   },
   "dependencies": {
-    "meow": "^3.7.0"
+    "yargs": "^3.3.0"
```
Comment (paulirish) [inline package.json]: Let's pin this directly to 3.3.0. We prefer our non-dev-deps arent moving.

Comment (brendankenny) [inline package.json]: oops, should have been 3.30.0

Comment (wardpeet) [inline package.json]: Looks like nothing to do here for me :p
